client/console/src/main/java/org/apache/syncope/client/console/rest/AuditHistoryRestClient.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.syncope.client.console.rest;
+
+import org.apache.syncope.common.lib.to.AuditEntryTO;
+import org.apache.syncope.common.lib.types.AuditElements;
+import org.apache.syncope.common.rest.api.beans.AuditQuery;
+import org.apache.syncope.common.rest.api.service.AuditService;
+import org.apache.wicket.extensions.markup.html.repeater.util.SortParam;
+
+import java.util.List;
+
+public class AuditHistoryRestClient extends BaseRestClient {
+    private static final long serialVersionUID = -381814125643246243L;
+
+    public List<AuditEntryTO> search(final String key,
+                                     final int page,
+                                     final int size,
+                                     final SortParam<String> sort,
+                                     final List<String> events,
+                                     final AuditElements.Result result) {
+        AuditQuery query = new AuditQuery.Builder(key)
+            .size(size)
+            .page(page)
+            .events(events)
+            .result(result)
+            .orderBy(toOrderBy(sort))
+            .build();
+        return getService(AuditService.class).search(query).getResult();
+    }
+
+    public List<AuditEntryTO> search(final String key,
```

Comment:
Why is this non-paging method required?

Reply:
I think this is used to populate the drop-down on the modal dialog with history items to compare; a similar approach seems to be [used here](https://github.com/apache/syncope/blob/2_1_X/client/console/src/main/java/org/apache/syncope/client/console/panels/HistoryConfList.java#L52-L58).

Reply:
I see, thanks.
Anyway, as I don't think that an infinite drop-down is applicable anyway, can't we just always used the paged method display only the last N items, where N could be sensibly set to, say, 15/20?

About the existing method reference, think we are going to refactor / cleanup that too.

Reply:
Sounds good, sure; I'll make some adjustments.